apps/studio/src/lib/models.ts

```diff
@@ -10,4 +10,6 @@ export enum EditorMode {
     PAN = 'pan',
     INSERT_TEXT = 'insert-text',
     INSERT_DIV = 'insert-div',
+    INSERT_MAIN_BUTTON = 'insert-main-button',
```

Comment:
Instead of adding a new mode for each new element type which does not scale. What are your thoughts on grouping it all  (besides insert text because that has custom behavior) under something like `INSERT_ELEMENT`, then within that mode, we have a "sub-mode" which signifies what type of element it is?

Reply:
Sounds like a good idea to me! I'll give it a go!